### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by rickecon**
This PR updates the documentation for `OG-UK`, including fixing the references to the United States, U.S. data, and the `Tax-Calculator` microsimulation model as noted in Issue #44. This PR does the following.

* Fixes typos and makes small stylistic changes
* Change references from `OG-USA` to `OG-UK`
* Get rid of references to `OG-UK-Calibration`
* Add `OGUK_references.bib`, `citations.bib`, `OGUK_references.md`, and `citations.md`, and add those two `.bib` files in the `_config.yml` `bibtex_bibfiles:` entry
* In `_config.yml`, update the copyright year to 2022, the repo URL, and the `config:` settings for `BibTeX` and `mathjax`
* Update the `_toc.yml`
* Fix API documentation in `_toc.yml`, `public_api.rst`, `calibrate.rst`, `get_micro_data.rst`
* Update `intro.md`, `contributor_guide.md`, and `tax_functions.md`

cc: @jdebacker @nikhilwoodruff

### CODE REVIEW

**Comment by jdebacker** [comment]
Thanks @rickecon!

**Comment by rickecon** [comment]
@jdebacker. This PR is now ready for review. Since the first set of documentation commits described in the first comment of this thread, I made the following changes to the GH Action `.yml` files:
* The `has-functional-changes.sh` and `is-version-number-acceptable.sh` shell scripts referred to a `CONTRIBUTING.md` file that doesn't exist. I changed the reference to `CHANGELOG.md`.
* Added current version 0.0.2 to the `CHANGELOG.md`.
* Removed the Python versions 3.8 and 3.9 from `setup.py`.
* Commented out the `test_frs()` test in the `test_get_micro_data.py` module because it was failing in the `get_micro_data.get_calculator_output()` method. We should fix this in the future.
* Disaggregated the GH actions from the two `push.yaml` and `pr.yml` files into the current `build_and_test.yaml`, `check_black.yml`, `deploy_docs.yml`, `docs_check.yml`, and `version_check.yml`
* Fixed some black formatting in `examples/plot.py` and `get_micro_data.py`

Regarding the current set of CI tests:
* The black and docs build tests are passing.
* The check versioning test is failing because I did not make a new version. However, I don't think this documentation and CI PR deserves a new version.
* The build and test tests are now passing because I commented out the `test_frs()` test in the `test_get_micro_data.py` module. This test was failing because of an issue loading the FRS data in the `get_micr-Data.get_calculator_output()` method. We should fix this in the future.

File: .github/has-functional-changes.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #! /usr/bin/env bash
 
-IGNORE_DIFF_ON="README.md CONTRIBUTING.md Makefile .gitignore LICENSE* .github/**"
+IGNORE_DIFF_ON="README.md CHANGELOG.md Makefile .gitignore LICENSE* .github/** ./docs/*"
```
Comment (nikhilwoodruff) [inline .github/has-functional-changes.sh]: Could add the test folders?

**Comment by rickecon** [comment]
@jdebacker @nikhilwoodruff. This PR is ready for final review once all tests pass (except for versioning and codecov).

**Pull request merged**